### PR TITLE
[stable/datadog] add configurable affinity and tolerations

### LINKED ISSUE #5382: [stable/datadog]  add configurable affinity and tolerations
**Is this a request for help?**:

---
FEATURE REQUEST

**Which chart**: stable/datadog

Unlike daemonsets, deployments in this chart do not have tolerations and affinities. Still, those can be useful in some cases. 
Here is mine: datadog can monitor etcd out-of-the-box, so I think I can use it to monitor my etcd cluster used by my kubernetes. But I need a secure way to pass kubernetes certificates to datadog. To me, it is okay to mount kubernetes directory inside a datadog agent pod, so I need something like a single-replica deployment to be able to schedule a pod on my master.

### PR DESCRIPTION
**Pull request by ivanfitenko**
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #5382

### CODE REVIEW

File: stable/datadog/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: datadog
-version: 0.11.2
+version: 0.11.3
```
Comment (timstoop) [inline stable/datadog/Chart.yaml]: As you're adding new features, please bump the minor version of the Chart.

File: stable/datadog/values.yaml [hunk 2]
```diff
@@ -47,6 +47,12 @@ daemonset:
 deployment:
   enabled: false
   replicas: 1
+  # Affinity for pod assignment
+  # Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+  # affinity: {}
```
Comment (timstoop) [inline stable/datadog/values.yaml]: I don't think this line needs to be commented out.

File: stable/datadog/values.yaml [hunk 3]
```diff
@@ -47,6 +47,12 @@ daemonset:
...
+  # Tolerations for pod assignment
+  # Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  # tolerations: []
```
Comment (timstoop) [inline stable/datadog/values.yaml]: Same for this one. As we don't do it in other Charts either, even though the value is empty.

**Review by timstoop — CHANGES_REQUESTED** [review]

**Comment by irabinovitch** [comment]
@ivanfitenko most of the datadog maintainers are at kubecon this week, but we'll get back to you on the review ASAP.

File: stable/datadog/Chart.yaml (hunk 1 above)
Comment (unguiculus) [reply stable/datadog/Chart.yaml]: Yes, minor bump, please

**Review by hkaj — CHANGES_REQUESTED** [review]
hi @ivanfitenko 
sorry for the delay here. This looks good to me, can you please bump the version to `0.14` so we can merge it?

**Comment by ivanfitenko** [comment]
@hkaj done

**Comment by hkaj** [comment]
thanks @ivanfitenko :)

**Pull request merged**